public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.tsx

```diff
@@ -10,17 +10,43 @@ import { PromQuery } from '../../types';
 import { promQueryModeller } from '../PromQueryModeller';
 import { buildVisualQueryFromString } from '../parsing';
 import { FeedbackLink } from '../shared/FeedbackLink';
-import { QueryEditorModeToggle } from '../shared/QueryEditorModeToggle';
+import { QueryEditorModeToggle, UIOptions as ModeToggleUIOptions } from '../shared/QueryEditorModeToggle';
 import { QueryHeaderSwitch } from '../shared/QueryHeaderSwitch';
 import { QueryEditorMode } from '../shared/types';
 import { changeEditorMode, getQueryWithDefaults } from '../state';
 
 import { PromQueryBuilderContainer } from './PromQueryBuilderContainer';
 import { PromQueryBuilderExplained } from './PromQueryBuilderExplained';
-import { PromQueryBuilderOptions } from './PromQueryBuilderOptions';
+import { PromQueryBuilderOptions, UIOptions as OptionsUIOptions } from './PromQueryBuilderOptions';
 import { PromQueryCodeEditor } from './PromQueryCodeEditor';
 
-export const PromQueryEditorSelector = React.memo<PromQueryEditorProps>((props) => {
+interface UIOptions {
```

Comment:
Opted for adding options to the editor to show/hide different parts. Would rather compose smaller parts into different editor but right now the structure does not allow that easily so this seemed quicker.